scipy/signal/waveforms.py

```diff
@@ -479,3 +479,55 @@ def _sweep_poly_phase(t, poly):
     intpoly = polyint(poly)
     phase = 2 * pi * polyval(intpoly, t)
     return phase
+
+
+def delta(n, i=0):
+    """
+    Return a unit impulse waveform (Kronecker delta).
+
+    Parameters
+    ----------
+    n : int
+        Number of samples in the waveform
+    i : float, optional
+        Index at which the value is 1.  Defaults to the 0th element.
+        If value is an integer, output is dtype int.
+        If value is non-integer, output is float, and uses `sinc` to
+        calculate the ideally bandlimited impulse function.
+
+    Returns
+    -------
+    y : ndarray
+        Output array containing an impulse waveform.
+
+    Notes
+    -----
+    For float indices, this can also be considered the (truncated) impulse
+    response of an ideal fractional delay.
+
+    Examples
+    --------
+    An impulse at the 0th element:
+
+    >>> from scipy import signal
+    >>> signal.delta(8)
+    array([1, 0, 0, 0, 0, 0, 0, 0])
+
+    Centered:
+
+    >>> signal.delta(7, 7//2)
+    array([0, 0, 0, 1, 0, 0, 0])
+
+    Interpolated:
+
+    >>> signal.delta(7, 7/2.)
+    array([-0.0909,  0.1273, -0.2122,  0.6366,  0.6366, -0.2122,  0.1273])
```

Comment:
It would be cool to add to this example how this could equivalently be done by taking `fft(signal.delta(7))`, multiplying it by a complex exponential (to get a fractional time-delay), taking the `ifft`, and showing that the result is identical (to precision).